tests/unit/lms/resources/_js_config_test.py

```diff
@@ -0,0 +1,159 @@
+
+    def test_it_is_None_for_non_lti_users(self, context, pyramid_request):
+        pyramid_request.lti_user = None
+
+        assert JSConfig(context, pyramid_request).config["authToken"] is None
+
+    @pytest.fixture
+    def authToken(self, config):
+        return config["authToken"]
+
+
+class TestJSConfigHypothesisClient:
+    """Unit tests for the "hypothesisClient" sub-dict of JSConfig."""
+
+    def test_it_contains_one_service_config(self, config):
+        assert len(config["services"]) == 1
+
+    def test_it_includes_the_api_url(self, config):
+        assert config["services"][0]["apiUrl"] == "https://example.com/api/"
+
+    def test_it_includes_the_authority(self, config):
+        assert config["services"][0]["authority"] == "TEST_AUTHORITY"
+
+    def test_it_disables_share_links(self, config):
+        assert config["services"][0]["enableShareLinks"] is False
+
+    def test_it_includes_grant_token(self, config):
+        before = int(datetime.datetime.now().timestamp())
+
+        grant_token = config["services"][0]["grantToken"]
+
+        claims = jwt.decode(
+            grant_token,
+            algorithms=["HS256"],
+            key="TEST_JWT_CLIENT_SECRET",
+            audience="example.com",
+        )
+        after = int(datetime.datetime.now().timestamp())
+        assert claims["iss"] == "TEST_JWT_CLIENT_ID"
+        assert claims["sub"] == "acct:example_username@TEST_AUTHORITY"
+        assert before <= claims["nbf"] <= after
+        assert claims["exp"] > before
+
+    def test_it_includes_the_group(self, config):
+        groups = config["services"][0]["groups"]
+
+        assert groups == ["example_groupid"]
+
+    def test_it_is_empty_if_provisioning_feature_is_disabled(
+        self, context, pyramid_request
+    ):
+        context.provisioning_enabled = False
+
+        assert JSConfig(context, pyramid_request).config["hypothesisClient"] == {}
+
+    def test_it_is_mutable(self, config):
+        config.update({"a_key": "a_value"})
+
+        assert config["a_key"] == "a_value"
+
+    @pytest.fixture
+    def config(self, config):
+        return config["hypothesisClient"]
+
+
+class TestJSConfigRPCServer:
+    """Unit tests for the "rpcServer" sub-dict of JSConfig."""
+
+    def test_it(self, config):
+        assert config == {"allowedOrigins": ["http://localhost:5000"]}
+
+    @pytest.fixture
+    def config(self, config):
+        return config["rpcServer"]
+
+
+class TestJSConfigURLs:
+    """Unit tests for the "urls" sub-dict of JSConfig."""
+
+    def test_it(self, config):
+        assert config == {}
+
+    @pytest.fixture
+    def config(self, config):
+        return config["urls"]
+
+
+@pytest.fixture(autouse=True)
+def BearerTokenSchema(patch):
+    return patch("lms.resources._js_config.BearerTokenSchema")
+
+
+@pytest.fixture
+def bearer_token_schema(BearerTokenSchema):
+    return BearerTokenSchema.return_value
```

Comment:
I can understand the desire to autouse `BearerTokenSchema` but I don't really get why `bearer_token_schema` is here specifically / exists at all.

Reply:
Fixtures like this `bearer_token_schema` one are nowt but a convenience, to avoid tests having to say `BearerTokenSchema.return_value` all the time.

It avoids tests having to repeatedly say things like `BearerTokenSchema.return_value.authorization_param.assert_called_once_with(...)` or
`BearerTokenSchema.return_value.authorization_param.return_value`. Or tests having to assign their own local variable `bearer_token_schema = BearerTokenSchema.return_value`.

It originated as an aid to understanding as we had developers on the team who found `.return_value` difficult to understand, especially when there were multiple chained `.return_value`'s.

But I think it also just makes the test code nicer and more readable as `bearer_token_schema` is shorter than `BearerTokenSchema.return_value`.

It's used consistently throughout our tests: whenever there's a `FooBar` fixture that will be a mocked class, and the `foo_bar` fixture below it will be the mocked object instance of that class.

The only reason `bearer_token_schema` is located at module level is to keep it right nextdoor to its close cousin `BearerTokenSchema`.